### PR TITLE
Bluetooth: Host: add appearance values

### PR DESCRIPTION
**Pull request by yang325**
Appearance values added for common scenarios.

### CODE REVIEW

**Comment by hermabe** [comment]
Would it be better to have defines for the categories and the subcategories and then a macro to combine them into the actual value? If one want to specify a subcategory one would anyway have to do some math with the current change, as only the base "generic" values are defined.

**Comment by yang325** [comment]
> Would it be better to have defines for the categories and the subcategories and then a macro to combine them into the actual value? If one want to specify a subcategory one would anyway have to do some math with the current change, as only the base "generic" values are defined.

It seems that the actual value matters. If someone wants to add a new appearance, one would check website (https://www.bluetooth.com/specifications/assigned-numbers/) and get the value directly.

**Comment by Thalley** [comment]
> Appearance values added for common scenarios.

Is there a reason why not all values are defined?

**Comment by yang325** [comment]
> > Appearance values added for common scenarios.
> 
> Is there a reason why not all values are defined?

In fact I follow the defines from Bluedroid. I think it could cover most cases.

**Comment by Thalley** [comment]
> 



> > > Appearance values added for common scenarios.
> > 
> > 
> > Is there a reason why not all values are defined?
> 
> In fact I follow the defines from Bluedroid. I think it could cover most cases.

Bluedruid => Fluoride, right? 

In any case, it sounds odds that Android only uses a subset, but IMO if we are adding a block of assigned numbers (like the appearance value), we might as well add them all. Adding a subset just kind of seems odd. Don't know if others agree with that.

**Comment by hermabe** [comment]
> Adding a subset just kind of seems odd. Don't know if others agree with that.

I agree that we should add all of them. Wither by doing the calculations ourselves or giving the application a macro to calculate the value, eg. 
```c
#define APPEARANCE(cat, sub) ((cat << 6) | (sub & 0x3F))
```

**Comment by Thalley** [comment]
> > Adding a subset just kind of seems odd. Don't know if others agree with that.
> 
> I agree that we should add all of them. Wither by doing the calculations ourselves or giving the application a macro to calculate the value, eg.
> 
> ```c
> #define APPEARANCE(cat, sub) ((cat << 6) | (sub & 0x3F))
> ```

We don't really need to do the calculations ourselves; the Assigned Numbers document contains the "final" value for all defined appearances. 

I like the idea of using a macro or even something like

```c
struct bt_appearance {
    uint16_t sub_category : 6;
    uint16_t category : 10;
}
```

The downside of supplying the macro or struct, is that users can use existing defined category and subgategories in combinations that is not defined. 

E.g. category `0x003` (Watch) is only defined with subgategory `0x00C0` (Generic Watch), `0x00C1` (Sports Watch) and `0x00C2` (Smartwatch), but could perhaps easily (and incorrectly) be used with `0x0086` (Wearable computer (watch size). 

The Assigned Numbers page has a table for which subgategory values are allowed for which categories, and with a macro/struct we cannot provide any help to the user. We could of reject any appearance in the API that isn't valid. 

In the end, I think that defining all existing appearance values with their value, as defined by the Assigned Numbers document, is likely the best solution here. We will of course need to revisit these values, like any other value in the assigned numbers document, occasionally to keep them up to date (to add new ones).

**Comment by yang325** [comment]
May I clarify it?
1. We need to include all appearances defined in the document.
2. No macro or something else needed as it's NOT very useful. The final values are what we need.

**Comment by yang325** [comment]
@hermabe @Thalley Done.

File: include/zephyr/bluetooth/gap.h [hunk 1]
```diff
@@ -78,6 +78,309 @@ extern "C" {
 #define BT_LE_AD_GENERAL                0x02 /* General Discoverable */
 #define BT_LE_AD_NO_BREDR               0x04 /* BR/EDR not supported */
 
+/* Appearance Values Last Modified on 2023-01-05 */
+#define BT_APPEARANCE_UNKNOWN                          0x0000 /* Generic Unknown */
```
Comment (Thalley) [inline include/zephyr/bluetooth/gap.h:82]: Not something we've done a lot of yet, but we could consider moving these to a `enum bt_appearance` and use that new enum type in `bt_get_appearance` and `bt_set_appearance`. However those will require a (non-breaking) stable API change, so maybe best to leave for another PR. 

@hermabe Thoughts?

Comment (hermabe) [reply include/zephyr/bluetooth/gap.h:82]: Yeah, I think an enum is generally better, but that can be done in a different PR.

**Review by Thalley — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

**Review by hermabe — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]
